### PR TITLE
funarray:0.2.0 & ouset:0.1.0

### PR DESCRIPTION
**Pull request by ludwig-austermann**
I am submitting
- [x] a new package
- [ ] an update for a package

I have read and followed the submission guidelines and, in particular, I
- [x] selected a name in conformance with the guidelines
- [x] added a `typst.toml` file with all required keys
- [x] added a `README.md` with documentation for my package
- [x] have chosen a license and added a `LICENSE` file or linked one in my `README.md`
- [x] ensured that my submission does not infringe upon the rights of a third party
- [x] tested my package locally on my system and it worked
- [x] named this PR as `name:version` of the submitted package
- [x] agree that my package will not be removed without good reason

Description:
- `funarray` gives the user additional functions for functional programming. Apart from a simple implementation, various checks have been added for better coding experience and the possibility to use the unsafe equivalents for better performance.
- `ouset` provides typst equivalents of latex overset and underset functions. Spacing can easily be tweaked for alignment in equations.

### CODE REVIEW

File: packages/preview/funarray-0.2.0/typst.toml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+[package]
+name = "funarray"
+version = "0.2.0"
+entrypoint = "funarray.typ"
+authors = ["Ludwig Austermann"]
+license = "MIT"
+description = "A package providing convinient functional functions for typst to use on arrays."
+repository = "https://ludwig-austermann/typst-funarray"
```
Comment (istudyatuni) [inline packages/preview/funarray-0.2.0/typst.toml:8]: Link not working, probably you forgot `github.com/` ?

Comment (ludwig-austermann) [reply packages/preview/funarray-0.2.0/typst.toml]: Yes, you are right, thanks. The urls should now be fixed for both packages.

**Review by laurmaedje — COMMENTED** [review]
I left a few optional suggestions for `funarray`. Some also apply to `ouset`.

File: packages/preview/funarray-0.2.0/README.md [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+This functions groups according to a predicate into maximally sized chunks, where all elements have the same predicate value.
+
+```typst
+let f = (0,0,1,1,1,0,0,1)
+let g = group_by(f, x => x == 0)
+g = ((0, 0), (1, 1, 1), (0, 0), (1,))
+```
+
+### flatten
+Typst has a `flatten` method for arrays, however that method acts recursively. For instance
```
[the remark this answers is not in the document]
Comment (ludwig-austermann) [reply packages/preview/funarray-0.2.0/README.md:99]: Yes, or maybe `max-depth`?

File: packages/preview/funarray-0.2.0/typst.toml [hunk 3]
```diff
@@ -0,0 +1,8 @@
+[package]
+name = "funarray"
+version = "0.2.0"
+entrypoint = "funarray.typ"
+authors = ["Ludwig Austermann"]
+license = "MIT"
+description = "A package providing convinient functional functions for typst to use on arrays."
```
[the remark this answers is not in the document]
Comment (ludwig-austermann) [reply packages/preview/funarray-0.2.0/typst.toml]: Should be shorter now.

File: packages/preview/funarray-0.2.0/README.md [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+The cycle function concatenates the array to itself until it has a given size.
+
+```typst
+let c = cycle(range(5), 8)
+c = (0, 1, 2, 3, 4, 0, 1, 2)
+```
+
+Note that there is also the functionality to concatenate with `+` and `*` in typst.
+
+### windows and circular_windows
```
[the remark this answers is not in the document]
Comment (ludwig-austermann) [reply packages/preview/funarray-0.2.0/README.md]: Thanks, yes, consistency is very nice. Should be changed now.

File: packages/preview/funarray-0.2.0/README.md [hunk 5]
```diff
@@ -0,0 +1,126 @@
+# FunArray Package
+This package provides some convinient functional functions for [typst](https://typst.app/) to use on arrays.
+
+## Usage
+All relevant code sits inside `funarray.typ`. Some examples and also the functions part of this README, are given as `.typ + .pdf` files in the examples folder.
```
[the remark this answers is not in the document]
Comment (ludwig-austermann) [reply packages/preview/funarray-0.2.0/README.md]: Ok, changed it

**Comment by laurmaedje** [comment]
Thank you!

**Pull request merged**